### DISCUSSION TITLE
New devices - '_TZE200_zion52ef' and '_TZE200_fhn3negr'

### DISCUSSION THREAD

**Discussion by MarekCoufal** [Koenkk/zigbee2mqtt#17972] [Devices]
Good day,
I have a thermostatic head reporting as "_TZE200_zion52ef" and the other "_TZE200_fhn3negr". I process the data only through an external converter, because the current version of Z2M does not support it. What should I find out about the header and pass it on to someone here so that it is already supported in the next version?

**Comment by MarekCoufal** [comment]
My external convertor:
`
const legacy = require('zigbee-herdsman-converters/lib/legacy');
const fz = {...require('zigbee-herdsman-converters/converters/fromZigbee'), legacy: require('zigbee-herdsman-converters/lib/legacy').fromZigbee};
const tz = {...require('zigbee-herdsman-converters/converters/toZigbee'), legacy: require('zigbee-herdsman-converters/lib/legacy').toZigbee};
const exposes = {...require('zigbee-herdsman-converters/lib/exposes'), legacy: require('zigbee-herdsman-converters/lib/legacy').exposes};
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = {...require('zigbee-herdsman-converters/lib/tuya'), legacy: require('zigbee-herdsman-converters/lib/legacy').tuya};
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');

const e = exposes.presets;
const ea = exposes.access;

// Vacation [19,1,1,0,0,42,0,0]
//0 - Start Year (0x00 = 2000)
//1 - Start Month
//2 - Start Day
//3 - Start Hour
//4 - Start Minute
//5 - Temperature (1~59 = 0.5~29.5°C (0.5 step))
//6-7 - Duration in Hours (0~2400 (100 days))

// byte 0 - Day of Week (0~7 = Mon ~ Sun) <- redundant?
// byte 1 - 1st period Temperature (1~59 = 0.5~29.5°C (0.5 step))
// byte 2 - 1st period end time (1~96 = 0:15 ~ 24:00 (15 min increment, i.e. 2 = 0:30, 3 = 0:45, ...))
// byte 3 - 2nd period Temperature
// byte 4 - 2nd period end time
// ...
// byte 16 - 8th period end time
// byte 17 - 9th period Temperature

const LocaldataPoints = {
//    moesHold: 2,
    moesTempManualMode: 16,
//    moesLocalTemp: 24,
//    moesChildLock: 30,
//    moesBattery: 34,
    moesDayPresetTemp: 101,
    moesNightPresetTemp: 102,
    moesVacationPeriod: 103, //(raw, 8 bytes): Set Vacation Period and Temperature [19,1,1,0,0,42,0,0]
    moesTempOffset: 104,
//    moesTempAutoMode: 105,
    moesUnknown106: 106,
    moesUnknown107: 107,
    moesScheduleMonday: 109,
    moesScheduleTuesday: 110,
    moesScheduleWednesday: 111,
    moesScheduleThursday: 112,
    moesScheduleFriday: 113,
    moesScheduleSaturday: 114,
    moesScheduleSunDay: 115,
    moesOpenWindowTemp: 116,
    moesOpenWindowTime: 117,
    moesRapidModeCountdownTimer: 118,
    moesRequestUpdate: 120,
};

function DTT(Numbero) { //DataToTime
    Hod = Math.floor((Numbero*15)/60);
    Min = ((((Numbero*15)/60)-Hod)*60);
    if (Min == "0") { Min ="00"}
    return Hod + ':' + Min;
}

function Schedule(Pole) {
    TTX = '00:00';
    for (i = 1; i < 18; i++) {
        switch (i) {
            case 1: case 3: case 5: case 7: case 9: case 11: case 13: case 15: case 17:
                TTX = TTX + ' - ' + Pole[i]/2;
                break;
            case 2: case 4: case 6: case 8: case 10: case 12: case 14: case 16:
                TTX = TTX + ' - ' + DTT(Pole[i]);
                break;
        }
 if (Pole[i] === 96) { break; }
        
    }
    return TTX;
}


const fzLocal = {
        moes_thermostat: {
        cluster: 'manuSpecificTuya',
        type: ['commandGetData', 'commandSetDataResponse'],
        convert: (model, msg, publish, options, meta) => {
            const dp = msg.data.dp;
            const value = tuya.getDataValue(msg.data.datatype, msg.data.data);
                meta.logger.warn(`###################: XXXXX DP #${dp} DATA ${JSON.stringify(msg.data)} VALUE ${value}`);
//            meta.logger.warn(`### ${JSON.stringify(msg.data)} ###`);
            switch (dp) {
            case LocaldataPoints.moesHold:
                return {preset_mode: value ? 'manu' : 'auto', preset: value ? 'manu' : 'auto'};
            case LocaldataPoints.moesChildLock:
                return {child_lock: value ? 'lock' : 'unlock'};
            case LocaldataPoints.moesLocalTemp:
                meta.logger.warn(`###################: XXXXX DP #${dp} DATA ${JSON.stringify(msg.data)} VALUE ${value}`);
                return {local_temperature: parseFloat((value / 10).toFixed(1))};
//            case LocaldataPoints.moesTempAutoMode: //DP 105 teplota v AUTO
//                return {current_heating_setpoint_A: parseFloat((value / 2).toFixed(1))};
//            case LocaldataPoints.moesTempManualMode: //DP 16 teplota v MANU
//                return {current_heating_setpoint_M: parseFloat((value / 2).toFixed(1))};
//            case LocaldataPoints.moesBattery:
//                return {battery: value};
//                return {battery: value > 130 ? 100 : value < 70 ? 0 : ((value - 70)*1.7).toFixed(1)};
            case LocaldataPoints.moesScheduleMonday:
                return { '1-Monday': Schedule(value) };
            case LocaldataPoints.moesScheduleTuesday:
                return { '2-Tuesday': Schedule(value) };
            case LocaldataPoints.moesScheduleWednesday:
                return { '3-Wednesday': Schedule(value) };
            case LocaldataPoints.moesScheduleThursday:
                return { '4-Thursday': Schedule(value) };
            case LocaldataPoints.moesScheduleFriday:
                return { '5-Friday': Schedule(value) };
            case LocaldataPoints.moesScheduleSaturday:
                return { '6-Saturday': Schedule(value) };
            case LocaldataPoints.moesScheduleSunDay:
                return { '7-Sunday': Schedule(value) };
//            case LocaldataPoints.moesDayPresetTemp:
//                return { day_preset_temp: parseFloat((value / 2).toFixed(1))};
//            case LocaldataPoints.moesNightPresetTemp:
//                return { night_preset_temp: parseFloat((value / 2).toFixed(1))};
//            case LocaldataPoints.moesOpenWindowTemp:
//                return {open_window_temperature: parseFloat((value / 2).toFixed(1))};
//            case LocaldataPoints.moesOpenWindowTime:
//                return {open_window_time: value};
            case LocaldataPoints.moesTempOffset:
                return {temp_offset: value}
            case LocaldataPoints.moesRapidModeCountdownTimer:
                return {rapid_mode_countdown: value};
            case LocaldataPoints.moesUnknown106:
                return ;
            case  LocaldataPoints.moesUnknown107:
                return ;
            case  LocaldataPoints.moesRequestUpdate:
                return ;
            case LocaldataPoints.moesVacationPeriod:
                return {vacation: value};
            default:
                meta.logger.warn(`###################: NEZNAME DP #${dp} data ${JSON.stringify(msg.data)}`);
            }
        },
    },

}



const devices = [
    {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_zion52ef'}],
        model: 'TS0601_thermostat',
        vendor: 'TuYa',
        description: 'Radiator valve with thermostat',
        onEvent: tuya.onEventSetLocalTime,
        configure: tuya.configureMagicPacket,
        fromZigbee: [
            fz.ignore_basic_report,
            tuya.fz.datapoints,
            fz.ignore_tuya_set_time
        ],
        toZigbee: [
            tz.tuya_data_point_test, //pridano
        ],
               exposes: [
            ],
        meta: {
        // All datapoints go in here
        tuyaDatapoints: [
            [24, 'local_temperature', tuya.valueConverter.divideBy10], // Teplota
            [120, 'request_update', tuya.valueConverter.raw], // Vycitani
            [34, 'battery', tuya.valueConverterBasic.divideBy(1.28)], // Baterie
            [30, 'child_lock', tuya.valueConverter.lockUnlock], // ChildLock
            [2, 'preset', tuya.valueConverterBasic.lookup({'auto': 0, 'manu': 1})], // Mode
            [105, 'current_heating_setpoint_A', tuya.valueConverterBasic.divideBy(2)], // Teplota Auto
            [16, 'current_heating_setpoint_M', tuya.valueConverterBasic.divideBy(2)], // Teplota Manu
            [101, 'preset_temp_day', tuya.valueConverterBasic.divideBy(2)], // Teplota Den
            [102, 'preset_temp_night', tuya.valueConverterBasic.divideBy(2)], // Teplota Noc
            [116, 'open_window_temperature', tuya.valueConverterBasic.divideBy(2)], // Teplota otevrene okno
            [117, 'open_window_time', tuya.valueConverter.raw], // Otevrene okno cas
            [16, 'current_heating_setpoint_M', tuya.valueConverterBasic.divideBy(2)], // Teplota Manu

            
        ],
        },
    },
    {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_fhn3negr'}],
        model: 'TS0601_thermostat',
        vendor: 'TuYa',
        description: 'Radiator valve with thermostat',
        onEvent: tuya.onEventSetLocalTime,
        fromZigbee: [
            fz.ignore_basic_report,
            tuya.fz.datapoints,
            fz.ignore_tuya_set_time
        ],
        toZigbee: [
            tz.tuya_data_point_test, //pridano
        ],
               exposes: [
            ],
        meta: {
        // All datapoints go in here
        tuyaDatapoints: [
            [24, 'local_temperature', tuya.valueConverter.divideBy10], // Teplota
            [120, 'request_update', tuya.valueConverter.raw], // Vycitani
//            [34, 'Baterie', tuya.valueConverter.raw], // Baterie
            [34, 'battery', tuya.valueConverterBasic.divideBy(1.28)], // Baterie
            [30, 'child_lock', tuya.valueConverter.lockUnlock], // ChildLock
            [2, 'preset', tuya.valueConverterBasic.lookup({'auto': 0, 'manu': 1})], // Mode
            [105, 'current_heating_setpoint_A', tuya.valueConverterBasic.divideBy(2)], // Teplota Auto
            [16, 'current_heating_setpoint_M', tuya.valueConverterBasic.divideBy(2)], // Teplota Manu
            [101, 'preset_temp_day', tuya.valueConverterBasic.divideBy(2)], // Teplota Den
            [102, 'preset_temp_night', tuya.valueConverterBasic.divideBy(2)], // Teplota Noc
            [116, 'open_window_temperature', tuya.valueConverterBasic.divideBy(2)], // Teplota otevrene okno
            [117, 'open_window_time', tuya.valueConverter.raw], // Otevrene okno cas
            [16, 'current_heating_setpoint_M', tuya.valueConverterBasic.divideBy(2)], // Teplota Manu

            
        ],
        },
    },
    ];

module.exports = devices;
`

**Comment by Bro2111** [comment]
Hallo,
could you please send me your external converter for implementing the device. I hav also bought some thermostat valves. If I copy the converter above I get an error. 
Many thanks! Thomas

**Comment by MarekCoufal** [comment]
[external-devices.txt](https://github.com/user-attachments/files/18696892/external-devices.txt)
Zigbee2MQTT version 1.42.0
